node/src/effect.rs

```diff
@@ -1505,14 +1525,11 @@ impl<REv> EffectBuilder<REv> {
         futures::future::join_all(
             era_ids
                 .into_iter()
-                .filter_map(|era_id| {
-                    era_id
-                        .0
-                        .checked_sub(1)
-                        .map(|prev_era_num| (EraId(prev_era_num), era_id))
-                })
-                .map(|(prev_era_id, era_id)| {
-                    self.get_switch_block_at_era_id_from_storage(prev_era_id)
+                .filter(|era_id| *era_id != EraId(0)) /* we would get None for that era and that
+                would make it seem like the entire
+                function failed */
```

Comment:
Weird formatting. I'd just make that a `//` comment in the lines before the `filter` call.

Reply:
Sometimes `rustfmt` formats such comments nicely, and sometimes it doesn't - not sure why. I'll do it the way you're suggesting :+1: